ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Types.hs

```diff
@@ -595,6 +616,14 @@ assertPeerSelectionState PeerSelectionState{..} =
     -- the alternative is a record of failure, but these are not (yet) failed.
   . assert (Set.isSubsetOf establishedPeersSet (KnownPeers.availableToConnect knownPeers))
 
+    -- If node is in YoungEnough state then it can't have only BootstrapPeers
+  . assert (currentLedgerStateJudgement /= YoungEnough || not hasOnlyBootstrapPeers)
```

Comment:
What if we are in transition, we switched to `YoungEnough` but we haven't yet acquired non bootstrap peers.